src/Type/Php/MbStrlenFunctionReturnTypeExtension.php

```diff
@@ -0,0 +1,44 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type\Php;
+
+use PhpParser\Node\Expr\FuncCall;
+use PHPStan\Analyser\Scope;
+use PHPStan\Reflection\FunctionReflection;
+use PHPStan\Reflection\ParametersAcceptorSelector;
+use PHPStan\Type\Constant\ConstantBooleanType;
+use PHPStan\Type\Constant\ConstantStringType;
+use PHPStan\Type\IntegerType;
+use PHPStan\Type\Type;
+use PHPStan\Type\TypeUtils;
+
+class MbStrlenFunctionReturnTypeExtension implements \PHPStan\Type\DynamicFunctionReturnTypeExtension
+{
+
+	public function isFunctionSupported(FunctionReflection $functionReflection): bool
+	{
+		return $functionReflection->getName() === 'mb_strlen';
+	}
+
+	public function getTypeFromFunctionCall(FunctionReflection $functionReflection, FuncCall $functionCall, Scope $scope): Type
+	{
+		$returnType = ParametersAcceptorSelector::selectSingle($functionReflection->getVariants())->getReturnType();
+
+		if (count($functionCall->args) < 1) {
+			return $returnType;
+		}
+		if (!isset($functionCall->args[1])) {
+			return new IntegerType();
+		}
+
+		$result = array_unique(array_map(function (ConstantStringType $encoding): bool {
+			return @mb_strlen('', $encoding->getValue()) !== false; // @ = silence the undocumented warning
```

Comment:
This extension should first check that the extension is available, if not, just return `$returnType`. The extension should not crash in case this function is unknown 😊 I noticed this because the Composer requirements checker failed.

Reply:
Yep, I saw that too. Thing is, I'm gonna have to whitelist mb in the whole project, because requirements checker is stupid and won't see the `if (extension_loaded())`..

Reply:
I agree.